### ISSUE TITLE
Issue with [[SFRestAPI sharedInstance] sendRESTRequest:fileContentRequest failBlock:^(NSError *e) always fail

### ISSUE THREAD

**Issue by hiralnaik** [forcedotcom/SalesforceMobileSDK-iOS#1510]
Hello there

We have an app with salesforce SDK. Recently We upgraded to Salesforce SDK from 4.0.0 to 4.1.1. After upgrading, we are having issue with download any file from the Salesforce. It used to work with previous Salesforce iOS SDK. We didn't change any code in our classes. Can you please help me with it ? or Did I miss anything in my code ?

Here is the method name that We used to download.
[SFRestAPI sharedInstance] sendRESTRequest:fileContentRequest failBlock:^(NSError *e).

Here is my code.
```
SFRestRequest *fileContentRequest = [[SFRestAPI sharedInstance] requestForFileContents:Id version:nil];
        [[SFRestAPI sharedInstance] sendRESTRequest:fileContentRequest failBlock:^(NSError *e)
         {
         }
         completeBlock:^(NSData *fileResponseData)
         {
         }];
```

**Error**:
`Error Domain=CSFNetworkErrorDomain Code=404 "The requested resource does not exist" UserInfo={action=<SFRestAPISalesforceAction: 0x7d83b6c0, GET "/chatter/files/XXXXXXXXXXXXX/content" (0.00)>, NSLocalizedFailureReason=NOT_FOUND, NSLocalizedDescription=The requested resource does not exist, isAuthenticationFailure=false}`

Thanks for your help.

**Comment by wmathurin** [comment]
In 4.1.1, we change the default api version to v36.0 and it looks like the files end point changed (see https://developer.salesforce.com/docs/atlas.en-us.chatterapi.meta/chatterapi/connect_resources_files_content.htm).
If you change apiVersion on your SFRestAPI instance to v35.0, it should work again.

**Comment by hiralnaik** [comment]
Yes I changed kSFRestDefaultAPIVersion constant to "v35.0" and it is working fine. So what will be the option here since I can't modify the SDK ?

**Comment by wmathurin** [comment]
You should be able to do [SFRestAPI sharedInstance].apiVersion = @"v35.0"

**Comment by wmathurin** [comment]
We'll fix the SDK for our next point release.

**Comment by hiralnaik** [comment]
Thanks. I will do. Do you know when are you guys planning to release next sdk ?

**Comment by wmathurin** [comment]
We are shooting for June.

**Comment by hiralnaik** [comment]
Sounds good. Thanks for your help on it.
